tap_iterable/iterable.py

```diff
@@ -32,25 +32,25 @@ def _now(self):
 
   def _daterange(self, start_date, end_date):
     total_days = (utils.strptime_with_tz(end_date) - utils.strptime_with_tz(start_date)).days
+    remaining_days = int(total_days / self.api_window_in_days)
     if total_days >= self.api_window_in_days:
-      for n in range(int(total_days / self.api_window_in_days)):
-        yield (utils.strptime_with_tz(start_date) + n * timedelta(self.api_window_in_days)).strftime("%Y-%m-%d %H:%M:%S")
+      for n in range(remaining_days):
+        yield ((utils.strptime_with_tz(start_date) + n * timedelta(self.api_window_in_days)).strftime("%Y-%m-%d %H:%M:%S"))
+      if int(total_days % self.api_window_in_days) > 0 :
+        start_slot_date = (utils.strptime_with_tz(start_date) + remaining_days * timedelta(self.api_window_in_days)).strftime("%Y-%m-%d %H:%M:%S")
+        yield (start_slot_date)
     else:
-      yield start_date
-
-
-  def _get_end_datetime(self, startDateTime):
-    endDateTime = utils.strptime_with_tz(startDateTime) + timedelta(self.api_window_in_days)
-    return endDateTime.strftime("%Y-%m-%d %H:%M:%S")
+      yield utils.strptime_with_tz(start_date).strftime("%Y-%m-%d %H:%M:%S")
 
 
   def retry_handler(details):
-    logger.info("Received 429 -- sleeping for %s seconds",
+    LOGGER.info("Received 429 -- sleeping for %s seconds",
                 details['wait'])
 
-  @backoff.on_exception(backoff.expo,
+  @backoff.on_exception(backoff.constant,
```

Comment:
any specific reason to change exponential backoff with constant?

Reply:
Iterable has low api request limit i.e. 4 requests/minute and backoff time is of 1 minute.Because of which, exponential backoff logic was causing too many retries than successful api calls which was also impacting the tap performance. After some analysis, we identified that with constant backoff logic we can reduce the number of retries and eventually improve the performance